crates/sim2h/src/lib.rs

```diff
@@ -62,35 +95,114 @@ pub struct Sim2h {
 }
 
 impl Sim2h {
-    pub fn new(
-        crypto: Box<dyn CryptoSystem>,
-        stream_manager: StreamManager<std::net::TcpStream>,
-        bind_spec: Lib3hUri,
-    ) -> Self {
+    pub fn new(crypto: Box<dyn CryptoSystem>, bind_spec: Lib3hUri) -> Self {
+        let pool = Pool::new();
+        pool.push_job(Box::new(Arc::new(Mutex::new(Tick::new()))));
+
+        let (wss_send, wss_recv) = crossbeam_channel::unbounded();
+        let (msg_send, msg_recv) = crossbeam_channel::unbounded();
+
         let mut sim2h = Sim2h {
             crypto,
             bound_uri: None,
             connection_states: RwLock::new(HashMap::new()),
             spaces: HashMap::new(),
-            stream_manager,
+            pool,
+            wss_recv,
+            msg_send,
+            msg_recv,
+            open_connections: HashMap::new(),
             num_ticks: 0,
             rrdht_arc_radius_recalc: std::time::Instant::now(),
             missing_aspects_resync: std::time::Instant::now(),
         };
 
-        debug!("Trying to bind to {}...", bind_spec);
-        let url: url::Url = bind_spec.clone().into();
-        sim2h.bound_uri = Some(
-            sim2h
-                .stream_manager
-                .bind(&url)
-                .unwrap_or_else(|e| panic!("Error binding to url {}: {:?}", bind_spec, e))
-                .into(),
-        );
+        sim2h.priv_bind_listening_socket(url::Url::from(bind_spec).into(), wss_send);
 
         sim2h
     }
 
+    /// bind a listening socket, and set up the polling job to accept connections
+    fn priv_bind_listening_socket(
+        &mut self,
+        url: Url2,
+        wss_send: crossbeam_channel::Sender<TcpWss>,
+    ) {
+        let config = TcpBindConfig::default();
+        let config = TlsBindConfig::new(config).dev_certificate();
+        let config = WssBindConfig::new(config);
+        let listen: TcpWssServer = InStreamListenerWss::bind(&url, config).unwrap();
+        self.bound_uri = Some(url::Url::from(listen.binding()).into());
+        self.pool
+            .push_job(Box::new(Arc::new(Mutex::new(ListenJob::new(
+                listen, wss_send,
+            )))));
+    }
+
+    /// if our listening socket has accepted any new connections, set them up
+    fn priv_check_incoming_connections(&mut self) {
+        if let Ok(wss) = self.wss_recv.try_recv() {
+            let url: Lib3hUri = url::Url::from(wss.remote_url()).into();
```

Comment:
Still concerned over the use of lib3huri, url, and url2. Maybe it deserves a follow up issue in the tree.